widget/list.go

```diff
@@ -0,0 +1,415 @@
+package widget
+
+import (
+	"image/color"
+	"math"
+
+	"fyne.io/fyne"
+	"fyne.io/fyne/canvas"
+	"fyne.io/fyne/driver/desktop"
+	"fyne.io/fyne/internal/widget"
+	"fyne.io/fyne/layout"
+	"fyne.io/fyne/theme"
+)
+
+// List is a widget that pools list items for performance and
+// lays the items out in a vertical direction inside of a scroller
+// List requires that all items are the same size
+type List struct {
+	BaseWidget
+
+	Length         func() int
+	CreateItem     func() fyne.CanvasObject
+	UpdateItem     func(index int, item fyne.CanvasObject)
+	OnItemSelected func(index int, item fyne.CanvasObject)
+
+	background      color.Color
+	offsetY         int
+	previousOffsetY int
+	selectedItem    *listItem
+	selectedIndex   int
+	itemMin         fyne.Size
```

Comment:
`itemMin` is initialized in `CreateRenderer` and never updated. Consider this scenario: a list of items where each item has an icon and text. After the list is initialized and `CreateRenderer` is called, the theme gets changed to one with a larger `TextSize` and `InlineIconSize`. `itemMin` size is now too small

Reply:
This should be addressed now.